### ISSUE TITLE
Support Multiple Projects with different GO versions

### ISSUE THREAD

**Issue by rvjc** [Microsoft/vscode-go#1892]
To support multiple projects with different paths and environments, I generally launch VS Code via a simple project-specific script with custom GOROOT, GOPATH and PATH settings. 

Unfortunately the VS Code Go extension appears to be running in a separate process and is not picking up my project-specific environment variables.  This produces the following error notification:

**Cannot find "go" binary. Update PATH or GOROOT appropriately.**

This means that I have to change the GO version in my system path every time I change to a different project which is both painful and error-prone. We work on Windows machines, so I'm not sure if the same problem arises with other OSes.

On the plus side, the Go extension lets me customise GOROOT and GOPATH in the project workspace settings. However, there is no way to specify a project-specific path for the "go" binary.   

There are three possible solutions:

1. The best solution to cover all scenarios (and all extensions) is to modify VS Code to replicate all environment variables from its own process when launching extensions.

2. If this seems too drastic, a VS Code command line option could enable/disable this behaviour.

3. Include a workspace setting for the Go extension to specify a custom "go" binary.  Actually this would also supplement 1 or 2 since it would only apply to the operation of the extension itself.

Our team recently moved to VS Code and are hugely impressed with the Go support. However, this is a critical issue which I hope you will resolve in the very near future.

**Comment by rvjc** [comment]
BTW, interestingly, when a I launch a terminal from within VS Code, it picks up my project-specific environment variables; so I assume it would be a relatively minor matter to do likewise when initialising extensions.

**Comment by rvjc** [comment]
Actually, we solved the problem and, with great embarrassment, I must admit that it was at our end.
Turned out to be an incongruous little artefact from previous IDE.

How often does this happen just after you post a bug report? :-)

So, just to reassure anyone who comes across this post:

1. The Go Extension picks up the environment of parent VS Code process (as expected).
2. It is very feasible to support different projects with different Go versions on the same machine.
